server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1390,6 +1393,30 @@ int numPendingDeletes(Index index) {
         }
     }
 
+    private SlowLogFieldProvider loadSlowLogFieldProvider() {
+        List<? extends SlowLogFieldProvider> slowLogFieldProviders = pluginsService.loadServiceProviders(SlowLogFieldProvider.class);
+        return new SlowLogFieldProvider() {
```

Comment:
Couldn't decide if this is overkill or not. I reduce all loaded slow log field providers to a single one here. We currently only have one and it will most likely stay that way.

Reply:
I don't think it's overkill. I think encouraging other additions here by supporting the plumbing for N where all we need is 1 is a net positive. However, I would suggest to lightly manually test adding from a second source just to be sure. (looks correct to me)

Reply:
Added a unit test to cover several providers. See this commit https://github.com/elastic/elasticsearch/pull/105621/commits/4f9a98587b73f0dae99e9bb6fb1524a19ceaf7d4